### PR TITLE
opensearch: Aoss master

### PR DESCRIPTION
**Pull request by matthewfala**
Please see notes and add comments to : https://github.com/fluent/fluent-bit/pull/6448

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [ ] Attached [local packaging test](./packaging/local-build-all.sh) output showing all targets (including any new ones) build.

**Documentation**

- [ ] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

**Review by PettitWesley — APPROVED** [review]

**Comment by edsiper** [comment]
please fix the conflicts on tests/internal/fuzzers/signv4_fuzzer.c

**Comment by edsiper** [comment]
ping

**Comment by matthewfala** [comment]
Rebased

**Review by PettitWesley — APPROVED** [review]

File: plugins/out_opensearch/opensearch.c [hunk 1]
```diff
@@ -1063,6 +1063,11 @@ static struct flb_config_map config_map[] = {
      0, FLB_FALSE, 0,
      "External ID for the AWS IAM Role specified with `aws_role_arn`"
     },
+    {
+     FLB_CONFIG_MAP_STR, "aws_service_name", "es",
```
Comment (seaHour) [inline plugins/out_opensearch/opensearch.c:1067]: the service name for aws opensearch serverless is "aoss". "es" doesn't work.

Comment (PettitWesley) [reply plugins/out_opensearch/opensearch.c:1067]: This plugin is meant to be used with serverless and non-serverless. Non-serverless (es) has always been the default. This change allows you to configure something different.

Comment (matthewfala) [reply plugins/out_opensearch/opensearch.c:1067]: Please see our example here @seaHour https://github.com/aws-samples/amazon-ecs-firelens-examples/tree/mainline/examples/fluent-bit/amazon-opensearch-serverless

**Comment by edsiper** [comment]
minor change to get this merged, please fix the subjects for these commits:

- `es: ...` -> `out_es: ...`
- `opensearch: ...` -> `out_opensearch: ...`

**Pull request merged**